data-structures/hash-tables.md

```diff
@@ -39,11 +40,31 @@ Knowing how hash tables work will give you a deeper understanding of why hash ta
 
 ### Common Mistakes / Misconceptions
 
-What is the difference between a hash map and a hash table? The two are used interchangeably.
+- What is the difference between a hash map and a hash table? The two are used interchangeably.
+
+- When should I use an array instead of a hash table? If your keys are sequential integers.
+
+- When does a JavaScript object stop being a hash table? When a property is added as a function.
+
+- [hashmap implementation](https://gist.github.com/dtx/3791420)
+
+ ![Image result for difference between hashmap and hashtable](https://cdncontribute.geeksforgeeks.org/wp-content/uploads/hashsetVshashmap.png)
+
+
+### Guided Practice
+
+Let's understand how to make hash tables and hash maps using JavaScript.
+
+- [How to implement simple hash table?](https://medium.freecodecamp.org/how-to-implement-a-simple-hash-table-in-javascript-cb3b9c1f2997)
+- Here is the link for the video on [how to make hash tables using JavaScripts](<https://www.youtube.com/watch?v=VundFD_ccgE>)  
+- The another link for what are hash tables and how to code them is [Basics of Hash Tables](https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/tutorial/). It contains the code of how *different types of hashing* can be done by *coding* in JS.
+- [How to implement hash table in javascript](https://reactgo.com/hashtable-javascript/)
+- [Difference Between HashMap And HashTable](https://javahungry.blogspot.com/2014/03/hashmap-vs-hashtable-difference-with-example-java-interview-questions.html)
+- ![Image result for tutorials for hash tables in JS](https://www.guru99.com/images/c-sharp-net/052616_1306_CCollection13.png)
```

Comment:
Thoughtful graphic!  Can you find one in javascript, though?  This looks like java.

Reply:
Is it compulsory to have in JS? Because such an explanatory image is not available. Though if get I will immediately put it. Or can you suggest some?

Reply:
So may I remove it for now?

Reply:
Sure especially since you labeled it as JS